js/widgets/rhythmruler.js

```diff
@@ -1145,18 +1145,22 @@ function RhythmRuler () {
         // FIXME: rough workaround for #508, investigate reasons why
         // the backspace press doesn't work by default
         var numberInput = docById('dissectNumber');
-        numberInput.addEventListener('keydown', function (event) {
-            if (event.keyCode === BACKSPACE) {
-               numberInput.value = numberInput.value.substring(0, numberInput.value.length - 1);
-            }
-
+        numberInput.onkeydown =  function (event) {
+            if (event.keyCode == 46) {
+                numberInput.value = numberInput.value.substring(0, numberInput.value.length - 1);
+            }   
+        };
+        numberInput.oninput = function (event) {
             // Put a limit on the size (2 <--> 128).
-            if (numberInput.value < 2) {
-                numberInput.value = 2;
-            } else if (numberInput.value > 128) {
+            numberInput.onmouseout = function(){
+                if (numberInput.value < 2) {
+                    numberInput.value = 2;
+                } 
+            }
+            if (numberInput.value > 128) {
```

Comment:
I don't understand why this conditional is outside of the onmouseout function.

Reply:
I added this condition outside the onmouseout function to change the value instantly as soon as it crosses 128.

Reply:
Why would you not want to change the other value (< 2) instantly as well?

Reply:
Because as the user types a number<2 (1 For eg)  in the input field , oniput function gets trigger .
If we dont use onmouseout function then the value will automatically change to 2 and it will not allow the user to enter the number after 1 (eg 11,12 and so on).

Reply:
Got it.